app/services/GuaranteeBalanceService.scala

```diff
@@ -35,11 +37,19 @@ import scala.concurrent.{ExecutionContext, Future}
 class GuaranteeBalanceService @Inject() (actorSystem: ActorSystem,
                                          connector: GuaranteeBalanceConnector,
                                          mongoLockRepository: MongoLockRepository,
-                                         config: FrontendAppConfig
+                                         config: FrontendAppConfig,
+                                         sessionRepository: SessionRepository
 )(implicit ec: ExecutionContext)
     extends Logging {
 
-  def submitBalanceRequest()(implicit hc: HeaderCarrier, request: DataRequest[_]): Future[Either[HttpResponse, BalanceRequestResponse]] =
+  def submitRequestOrPollForResponse()(implicit hc: HeaderCarrier, request: DataRequest[_]): Future[Either[HttpResponse, BalanceRequestResponse]] =
+    request.userAnswers.get(BalanceIdPage) match {
+      case Some(balanceId: BalanceId) => pollForGuaranteeBalance(balanceId)
+      case None                       => submitBalanceRequest
+    }
+
+  def submitBalanceRequest()(implicit hc: HeaderCarrier, request: DataRequest[_]): Future[Either[HttpResponse, BalanceRequestResponse]] = {
+    logger.info("[GuaranteeBalanceService][submitBalanceRequest]")
```

Comment:
A bit picky but can we have a little message in here to accompany the file/method names